ethereum-listener/src/lib/syncHandler.js

```diff
@@ -0,0 +1,44 @@
+const get = require('lodash/get')
+const { StaticPool } = require("node-worker-threads-pool")
+const debug = require('debug')('ethereum-listener:lib:syncHandler')
+const getSyncStartAndEndBlocks = require('./getSyncStartAndEndBlocks')
+
+module.exports = async ({
+  models,
+  sequelize,
+  web3,
+}) => {
+  const { start, end } = await getSyncStartAndEndBlocks({ models, sequelize, web3, models })
+
+  if ( !(start <= end) ) {
+    console.warn(`bad block numbers, cannot sync: Start-> ${start}, End-> ${end}`)
+    return
+  }
+
+  try {
+    const staticPool = new StaticPool({
+      size: 4,
```

Comment:
This size should be parameterized -- we can also pass it via environment variable to have more control in deployment.

Reply:
Absolutely. Done.